File: code/HoChanny/week3/boj_1260.js
const input = require('fs').readFileSync('example.txt').toString().trim().split('\n');
const [N, M, V] = input[0].split(' ').map(Number);
const edges = input.slice(1).map(line => line.split(' ').map(e => +e));

function dfs(graph, start, visited) {
    visited[start] = true;
    process.stdout.write(start + ' ');

    for (const nextNode of graph[start]) {
        if (!visited[nextNode]) {
            dfs(graph, nextNode, visited);
        }
    }
}

function bfs(graph, start) {
    const visited = Array(N + 1).fill(false);
    const queue = [start];
    visited[start] = true;

    while (queue.length) {
        const currentNode = queue.shift();
        process.stdout.write(currentNode + ' ');

        for (const nextNode of graph[currentNode]) {
            if (!visited[nextNode]) {
                visited[nextNode] = true;
                queue.push(nextNode);
            }
        }
    }
}

function solution(N, edges, V) {
    const graph = Array.from({ length: N + 1 }, () => []);

    for (const edge of edges) {
        const [u, v] = edge;
        graph[u].push(v);
        graph[v].push(u);
    }

    for (const neighbors of graph) {
        neighbors.sort((a, b) => a - b);
    }

    const visited = Array(N + 1).fill(false);
    dfs(graph, V, visited);
    console.log();

    bfs(graph, V);
    console.log();
}

solution(N, edges, V);

File: code/jjuny-won/week3/boj_1260.py
import sys
from collections import deque

def dfs(v):
    #방문 요소 = 1
    visited[v] = 1 
    print(v,end= ' ')
    for i in range(1,n+1):
        if graph[v][i] ==1 and visited[i] ==0:
            dfs(i)    
            
def bfs(v):
    queue=[v] #(deque[start])
    visited2[v]=1
    while queue:
        v  = queue.pop(0)
        print(v,end=' ')
        for i in range(1,n+1):
            if (graph[v][i]==1 and visited2[i]==0):
                queue.append(i)
                visited2[i] =1

#입력받는 부분
n,m,v = map(int,input().split())

#행렬 만들기
graph = [[0]*(n+1) for _ in range(n+1)]


for i in range(m):
    a,b = map(int,input().split())
    graph[a][b] = graph[b][a] = 1
    

#방문 리스트 만들기
visited = [0]*(n+1)
visited2 = [0]*(n+1)
dfs(v)
print()
bfs(v)


